src/common/AllExperiments/Experiments_Live.cs

```diff
@@ -0,0 +1,126 @@
+                {
+                    var result = task.Result;
+                    var allFeatureFlags = result.GetFeatureVariables();
+
+                    var assignmentContext = result.GetAssignmentContext();
+                    var featureNameSpace = allFeatureFlags[0].KeySegments[0];
+                    var featureFlag = allFeatureFlags[0].KeySegments[1];
+                    var featureFlagValue = allFeatureFlags[0].GetStringValue();
+
+                    if (featureFlagValue == "alternate")
+                    {
+                        IsExperiment = true;
+                    }
+
+                    PowerToysTelemetry.Log.WriteEvent(new OobeVariantAssignmentEvent() { AssignmentContext = assignmentContext, ClientID = AssignmentUnit });
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError("Error getting variant assignments for experiment", ex);
+            }
+        }
+
+        public bool IsExperiment { get; set; }
+
+        private string? AssignmentUnit { get; set; }
+
+        private IVariantAssignmentRequest GetVariantAssignmentRequest()
+        {
+            string? clientID = string.Empty;
+
+            string workingDirectory = Environment.CurrentDirectory;
+            var exeDir = Directory.GetParent(workingDirectory)?.Parent?.Parent?.FullName;
+            string settingsPath = @"src\common\AllExperiments\settings.json";
+            if (exeDir != null)
+            {
+                string jsonFilePath = Path.Combine(exeDir, settingsPath);
+
+                string json = File.ReadAllText(jsonFilePath);
+                var jsonDictionary = JsonSerializer.Deserialize<Dictionary<string, object>>(json);
+                if (jsonDictionary != null)
+                {
+                    if (!jsonDictionary.ContainsKey("clientid"))
+                    {
+                        jsonDictionary.Add("clientid", string.Empty);
+                        jsonDictionary["clientid"] = Guid.NewGuid().ToString();
+                        string output = JsonSerializer.Serialize(jsonDictionary);
+                        File.WriteAllText(jsonFilePath, output);
+                    }
+
+                    clientID = jsonDictionary["clientid"]?.ToString();
+                    AssignmentUnit = clientID;
+                }
+            }
+
+            var attrNames = new List<string> { "FlightRing" };
+            var attrData = AnalyticsInfo.GetSystemPropertiesAsync(attrNames).AsTask().GetAwaiter().GetResult();
+
+            var flightRing = string.Empty;
+
+            foreach (KeyValuePair<string, string> attr in attrData)
+            {
+                flightRing = attr.Value;
+            }
+
+            return new VariantAssignmentRequest
+            {
+                Parameters =
+                {
+                    // TBD: Adding traffic filters to target region.
+                    { "flightRing", flightRing },
```

Comment:
Should "flightRing" match the value that's used in `attrNames`? There it started with the capital letter and here with the lowercase one.

Reply:
Would it be better style for me to name it as "FlightRing" and FlightRing even though clientid is lowercase? I named it with camel case here since that was the naming convention I used throughout my code.

Reply:
I was asking because it'd also be good to have a constant instead of a string literal. Or 2 constants in case those values are independent.